egs/wsj/s5/steps/cleanup/make_utterance_graph.sh

```diff
@@ -101,16 +103,38 @@ while read line; do
   # Compiles G.fst
   if [ $ngram_order -eq 1 ]; then
     echo $words > $wdir/text
-    cat $wdir/text | utils/sym2int.pl --map-oov $oov -f 1- $lang/words.txt | \
+    if $top_words_list; then
+      cat $wdir/text | utils/sym2int.pl --map-oov $oov -f 1- $lang/words.txt | \
+      steps/cleanup/make_utterance_fst.pl $top_words_list | fstcompile |\
+      fstarcsort --sort_type=ilabel > $wdir/G.fst || exit 1;
+    else
+      cat $wdir/text | utils/sym2int.pl --map-oov $oov -f 1- $lang/words.txt | \
       utils/make_unigram_grammar.pl | fstcompile |\
       fstarcsort --sort_type=ilabel > $wdir/G.fst || exit 1;
+    fi
   else
     echo $words | awk -v voc=$lang/words.txt -v oov="$oov_txt" '
       BEGIN{ while((getline<voc)>0) { invoc[$1]=1; } } {
       for (x=1;x<=NF;x++) {
       if (invoc[$x]) { printf("%s ", $x); } else { printf("%s ", oov); } }
       printf("\n"); }' > $wdir/text
-    ngram-count -text $wdir/text -order $ngram_order "$srilm_options" -lm - |\
+    cat $top_words_list | utils/int2sym.pl -f 2 $lang/words.txt | python -c '
+import sys, math
+probs = dict()
+for line in sys.stdin.readlines():
+  splits = line.strip().split()
+  probs[splits[1]] = float(splits[0])
+
+print ("\data\\")
+print ("ngram 1={0}".format(len(probs)))
+print ("\\1-grams:")
+for w,p in probs.iteritems():
+  print ("{0:f}\t{1}".format(math.log10(p), w))
+print ("\end\\")' | ngram -order 1 -lm - -write-lm $wdir/top_unigram.arpa.gz
+
+    ngram-count -text $wdir/text -order $ngram_order "$srilm_options" -lm $wdir/text_lm.arpa.gz
```

Comment:
Are you still using SRILM for what you're doing?  I thought you said you were manually deciding the bigram vs. unigram weights.
We can't have this type of script depending on SRILM, for license reasons; and I think using a language modeling toolkit is overkill; what you need could easily be done with a very short python script.